### PR TITLE
add scale cache test manual presubmit

### PR DESCRIPTION
**Pull request by BenTheElder**
https://github.com/kubernetes/kubernetes/pull/126448

### CODE REVIEW

**Comment by BenTheElder** [comment]
> === FAIL: config/tests/testgrids TestKubernetesProwInstanceJobsMustHaveMatchingTestgridEntries (0.01s)
>    config_test.go:510: Job pull-kubernetes-e2e-gce-pull-through-cache does not have a matching testgrid testgroup

... we should revisit this, I don't think presubmits should always have a testgrid, it works much better for periodic / postsubmit

**Comment by BenTheElder** [comment]
Added testgrid annotations

File: config/jobs/kubernetes/sig-cloud-provider/gcp/gcp-gce.yaml [hunk 1]
```diff
@@ -307,6 +307,88 @@ presubmits:
...
+      containers:
+        - volumeMounts:
+          - name: cache-secret
+            readOnly: true
+            mountPath: /etc/registry-auth
+          env:
+          - name: KUBERNETES_REGISTRY_PULL_THROUGH_HOST
+            value: https://us-central1-docker.pkg.dev/v2/k8s-infra-e2e-scale-5k-project/k8s-5k-scale-cache/
+          - name: KUBERNETES_REGISTRY_PULL_THROUGH_BASIC_AUTH_TOKEN_PATH
+            value: /etc/registry-auth/token
```
Comment (dims) [inline config/jobs/kubernetes/sig-cloud-provider/gcp/gcp-gce.yaml:349]: `token` comes from the key in the secret.

File: config/jobs/kubernetes/sig-cloud-provider/gcp/gcp-gce.yaml [hunk 2]
```diff
@@ -307,6 +307,88 @@ presubmits:
...
+            - --test_args=--ginkgo.skip=\[Driver:.gcepd\]|\[Slow\]|\[Serial\]|\[Disruptive\]|\[Flaky\]|\[Feature:.+\] --minStartupPods=8
+            - --timeout=80m # thinking about making this longer? don't! 80m is a hard cap, and should get down to no more than 60m.
+          image: gcr.io/k8s-staging-test-infra/kubekins-e2e:v20240725-1d8ea3e909-master
+          # TODO: increase CPU for the pull-kubernetes-e2e-gce, we spend a LONG time building kubernetes and it slows iteration
+          resources:
+            limits:
+              cpu: 8
+              memory: "14Gi"
+            requests:
+              cpu: 8
```
Comment (BenTheElder) [inline config/jobs/kubernetes/sig-cloud-provider/gcp/gcp-gce.yaml:387]: https://github.com/kubernetes/test-infra/pull/33159

File: config/jobs/kubernetes/sig-cloud-provider/gcp/gcp-gce.yaml (hunk 1 above)
Comment (BenTheElder) [reply config/jobs/kubernetes/sig-cloud-provider/gcp/gcp-gce.yaml:349]: yes. I'm not sure these env are winding up down through to configure-helper though, I don't see the job failing but I also don't see it pulling through any images. will have to look through ARTIFACTS when the run is complete

**Pull request merged**